crates/iron-remote-desktop/src/session.rs

```diff
@@ -0,0 +1,81 @@
+use wasm_bindgen::JsValue;
+use web_sys::{js_sys, HtmlCanvasElement};
+
+use crate::clipboard::ClipboardTransaction;
+use crate::error::IronError;
+use crate::input::InputTransaction;
+use crate::DesktopSize;
+
+pub trait SessionBuilder {
+    type Session: Session;
+    type IronError: IronError;
+
+    fn init() -> Self;
+    #[must_use]
+    fn username(&self, username: String) -> Self;
+    #[must_use]
+    fn destination(&self, destination: String) -> Self;
+    #[must_use]
+    fn server_domain(&self, server_domain: String) -> Self;
+    #[must_use]
+    fn password(&self, password: String) -> Self;
+    #[must_use]
+    fn proxy_address(&self, address: String) -> Self;
+    #[must_use]
+    fn auth_token(&self, token: String) -> Self;
+    #[must_use]
+    fn desktop_size(&self, desktop_size: DesktopSize) -> Self;
+    #[must_use]
+    fn render_canvas(&self, canvas: HtmlCanvasElement) -> Self;
+    #[must_use]
+    fn set_cursor_style_callback(&self, callback: js_sys::Function) -> Self;
+    #[must_use]
+    fn set_cursor_style_callback_context(&self, context: JsValue) -> Self;
+    #[must_use]
+    fn remote_clipboard_changed_callback(&self, callback: js_sys::Function) -> Self;
+    #[must_use]
+    fn remote_received_format_list_callback(&self, callback: js_sys::Function) -> Self;
+    #[must_use]
+    fn force_clipboard_update_callback(&self, callback: js_sys::Function) -> Self;
+    #[must_use]
+    fn extension(&self, value: JsValue) -> Self;
+    fn connect(&self) -> impl core::future::Future<Output = Result<Self::Session, Self::IronError>>;
```

Comment:
suggestion: I think `async fn` is supported now (for static dispatch).
```suggestion
    async fn connect(&self) -> Result<Self::Session, Self::IronError>;
```

Reply:
Done